elements/pfe-button/BaseButton.ts

```diff
@@ -127,27 +20,24 @@ export type ButtonVariant = (
 export abstract class BaseButton extends LitElement {
   static readonly styles = [styles];
 
+  static readonly formAssociated = true;
+
+  static readonly shadowRootOptions: ShadowRootInit = { ...LitElement.shadowRootOptions, delegatesFocus: true };
+
   /** Disables the button */
-  @observed
   @property({ reflect: true, type: Boolean }) disabled = false;
 
-  /**
-   * Changes the style of the button.
-   * - Primary: Used for the most important call to action on a page. Try to
-   *   limit primary buttons to one per page.
-   * - Secondary: Use secondary buttons for general actions on a page, that
-   *   don’t require as much emphasis as primary button actions. For example,
-   *   you can use secondary buttons where there are multiple actions, like in
-   *   toolbars or data lists.
-   * - Tertiary: Tertiary buttons are flexible and can be used as needed.
-   */
-  @property({ reflect: true }) variant: ButtonVariant = 'primary';
+  @property({ reflect: true }) type?: 'button'|'submit'|'reset';
```

Comment:
@heyMP @zeroedin does this need to be abstract, too?
case: `<some-button type="wacky">`

OR, do we say that `type` on a custom button should not ever be anything other than the built-in types?

Reply:
I don't think this should be abstracted.  It should align with the button type from the HTML spec.  I think it's good to stay.

Reply:
@bennypowers what would you think about setting `reflect:false`.  @zeroedin and I were having a discussion about removing those two-way data bindings.  Opting for internal states using `classMap()`.

Reply:
@heyMP [`HTMLButtonElement#type` reflects](https://codepen.io/bennyp/pen/PoarMvw?editors=1000)

Reply:
Of course 👍